### PR TITLE
move StreamJsonRpc 1.xxx to 2.xxx

### PR DESCRIPTION
**Pull request by heejaechang**
we made this move "https://github.com/dotnet/roslyn/pull/33960" and reverted with "https://github.com/dotnet/roslyn/pull/34654"

we reverted since 
1. there were RPS regressions since we were the first one that made the move. 
2. it required works from partner team which we could avoid if we wait until them to go in first.

now @AArnott did the move on the platform on (1), we shouldn't have RPS problem anymore.

for (2), if partner team can't make it for preview3, they agreed to do the work for us to move in preview 3. and master is now for preview 3.

...

this is a straight revert. but if we see issue due to change after the revert, I will make the fix as a separate commit in this PR.

### CODE REVIEW

**Comment by heejaechang** [comment]
tagging @sharwell @shyamnamboodiripad @jinujoseph

File: src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs [hunk 1]
```diff
@@ -63,8 +63,7 @@ public static async Task<RemoteHostClient> CreateAsync(Workspace workspace, bool
             _inprocServices = inprocServices;
             _remotableDataRpc = remotableDataRpc;
 
-            _rpc = new JsonRpc(new JsonRpcMessageHandler(stream, stream), target: this);
-            _rpc.JsonSerializer.Converters.Add(AggregateJsonConverter.Instance);
```
Comment (ivanbasov) [inline src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs]: >AggregateJsonConverter.Instance [](start = 47, length = 31)

You can remove AggregateJsonConverter.Instance from AggregateJsonConverter

File: src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj [hunk 2]
```diff
@@ -93,6 +93,13 @@
     <!--     
       Exclude dependencies that are not inserted from Roslyn repo.
       Note: can't use globs due to https://github.com/Microsoft/msbuild/issues/3791.
+      
+      these dependency should be removed once StreamJsonRpc adds a build target for
+      net472 - https://github.com/dotnet/roslyn/issues/34558
+      
+      <_Dependency Remove="Nerdbank.Streams"/>
+      <_Dependency Remove="System.IO.Pipelines"/>
+      <_Dependency Remove="System.ValueTuple"/>
```
Comment (ivanbasov) [inline src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: As @sharwell mentioned here [https://github.com/dotnet/roslyn/issues/34558](https://github.com/dotnet/roslyn/issues/34558), should we consume 2.1 and remove them?

File: src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs (hunk 1 above)
Comment (ivanbasov) [reply src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs]: I see you still use it in Extensions. Maybe you should move it there.

---
In reply to: [277095163](https://github.com/dotnet/roslyn/pull/34860#discussion_r277095163) [](ancestors = 277095163)

File: src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj (hunk 2 above)
Comment (ivanbasov) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: This is the only concern I have. Other than that, the PR looks good.

---
In reply to: [277095521](https://github.com/dotnet/roslyn/pull/34860#discussion_r277095521) [](ancestors = 277095521)

Comment (AArnott) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: StreamJsonRpc 2.1 isn't stable yet. But 2.0.167 has what you're looking for.

Comment (ivanbasov) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: Thank you, @AArnott ! If 2.0.167 does not have those extra references, let us move forward to 2.0.167 and remove them.

Comment (AArnott) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: 2.0.167 includes a net472 target and thus drops the System.ValueTuple reference. Nerdbank.Streams and System.IO.Pipelines references remain.

Comment (heejaechang) [reply src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj]: sure. I can move to newer version.

File: src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs (hunk 1 above)
Comment (heejaechang) [reply src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs]: @ivanbasov I am not sure what you meant by this?

Comment (ivanbasov) [reply src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs]: @heejaechang you removed some lines 
`_rpc.JsonSerializer.Converters.Add(AggregateJsonConverter.Instance);`
in different files.

All them refer to `AggregateJsonConverter.Instance`. As far as I understand, now `AggregateJsonConverter.Instance` is used just once. It is used by `Extensions` and can be moved there.

Comment (heejaechang) [reply src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs]: this one is shared by multiple projects as linked file. so moving whole code to Extension.cs make this file to big. so, I would prefer leaving things as it is for now.

**Comment by heejaechang** [comment]
https://github.com/dotnet/roslyn/issues/35654

**Comment by heejaechang** [comment]
we need to move Pythia to newer Pythia to fix this issue.

05/14/2019 21:23:39 : Error : Error creating Pythia.RemotePythiaService instance of CLR service module 'pythia': System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. HResult='-2146232828' 
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.ServiceHub.HostLib.ServiceManager.CreateAndConfigureService(String name, Type serviceModuleType, WrappedStream stream)
===InnerException===
System.MissingFieldException: Field not found: 'Microsoft.CodeAnalysis.Remote.ServiceHubServiceBase.Rpc'. HResult='-2146233071' 
   at Pythia.RemotePythiaService..ctor(Stream stream, IServiceProvider serviceProvider)

File: src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs [hunk 3]
```diff
@@ -58,8 +59,8 @@ public async Task InvokeAsync(string targetName, IReadOnlyList<object> arguments
 
             try
             {
-
-                await _rpc.InvokeWithCancellationAsync(targetName, arguments, cancellationToken).ConfigureAwait(false);
+                // convert to array until the bug is fixed - https://github.com/microsoft/vs-streamjsonrpc/issues/272
+                await _rpc.InvokeWithCancellationAsync(targetName, arguments?.ToArray(), cancellationToken).ConfigureAwait(false);
```
Comment (AArnott) [inline src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs]: You know the fix for this is likely to get inserted into VS today.

File: src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs [hunk 4]
```diff
@@ -200,7 +200,8 @@ public async Task<bool> TryInvokeAsync(string targetName, Solution solution, IRe
                 pooledObject.Object.Add(scope.SolutionInfo);
                 pooledObject.Object.AddRange(arguments);
 
-                await connection.Target.InvokeAsync(targetName, pooledObject.Object, cancellationToken).ConfigureAwait(false);
+                // convert to array until the bug is fixed - https://github.com/microsoft/vs-streamjsonrpc/issues/272
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs]: 📝 This fix shipped as v2.0.187

File: src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs (hunk 3 above)
Comment (AArnott) [reply src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs]: [VS insertion PR](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/181335?_a=overview).

Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs]: ah.. thank you. probably I can remove this workaround then.

**Pull request merged**